api/v1beta1/cloudstackcluster_types.go

```diff
@@ -39,13 +39,35 @@ type CloudStackIdentityReference struct {
 	Name string `json:"name"`
 }
 
+type Network struct {
```

Comment:
I'm surprised to see some of the attributes being used for the spec.  I'd imagined ZoneId, Zone.NetworkId and Zone.NetworkType to be status-only attributes.

Reply:
+1 @rejoshed do all these have to passed in from eks-a? Or just some of them? I thought the input was ambiguous and capc would itself determine if the input was a name or ID

Reply:
Can pass either or, but do need to specify the correct one. So, id needs to be the ACS ID and name has to match the name.